### PR TITLE
Add coaching sessions endpoint

### PR DESCRIPTION
**Pull request by calebbourg**
## Description
This PR adds the ability to fetch coaching sessions by:
- `coaching_relationship_id`
- `from_date`
- `to_date`

### Changes
* Adds `GET /coaching_sessions` endpoint
* Adds `coaching_session::find_by` with the ability to filter by the above query params
* Adds seeds for `coaching_sessions` records
* Adds rapidoc specification for `GET /coaching_sessions`

### Testing Strategy
Tested locally using rapidoc
Screen Shot 2024-07-02 at 8 58 50 AM

### Concerns
I may consider moving the `coaching_relationship_id` into an explicit path parameter (`GET /coaching_relationships/:coaching_relationship_id/coaching_sessions`) so that this endpoint will require scoping the list of `coaching_sessions` to a specific `coaching_relationship_id`.  I think we can brainstorm on this when we implement an authorizations strategy.

### CODE REVIEW

File: entity/src/coaching_sessions.rs
Comment (jhodapp) [inline entity/src/coaching_sessions.rs:14]: Great call, these shouldn't be optional.

File: entity_api/src/coaching_session.rs
Comment (jhodapp) [inline entity_api/src/coaching_session.rs:15]: @calebbourg I'm curious if there's a way of getting the key name as a string to compare to here, i.e. "coaching_relationship_id", "from_date", etc by adding these as constant string literals in entity/src/coaching_sessions.rs and referring to them here by constant name. If this is tricky and thus takes a lot of time, don't worry about it then. I was just thinking if it's straightforward we might as well make this change.

File: entity_api/src/coaching_session.rs
Comment (jhodapp) [inline entity_api/src/coaching_session.rs:121]: One suggestion for another test is to test for a date range that doesn't match any coaching sessions and should return and empty `vec<Model>`. This is something that a user can run into on the frontend, so it'd be good to test for it returning cleanly.

File: entity_api/src/lib.rs
Comment (jhodapp) [inline entity_api/src/lib.rs:21]: I'm curious if there's a specific reason you chose `NaiveDate` over `DateTime`? It seems that if you were to use `DateTime` instead, you would have a [more flexible string parser](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.parse_from_rfc3339) to call, and YYY-MM-DD is a valid ISO8601 date. The reason I noticed this is because I was working with the same date format on the frontend and was able to use the `ts-luxmon` library's `DateTime` type to parse ISO8601 dates.

Comment (jhodapp) [reply entity_api/src/lib.rs:21]: Although, I did just notice you're only using this locally for tests, so perhaps this doesn't matter. Your call on this.

File: web/src/controller/coaching_session_controller.rs
Comment (jhodapp) [inline web/src/controller/coaching_session_controller.rs:22]: Can we use `DateTime` instead of `NaiveDate`?

File: web/src/controller/coaching_session_controller.rs
Comment (jhodapp) [inline web/src/controller/coaching_session_controller.rs:37]: Is this comment still relevant?

**Review by jhodapp — CHANGES_REQUESTED** [review]
This looks so good Caleb! I left some questions and suggestions inline, take or leave any of them as you see fit.

File: entity_api/src/coaching_session.rs
Comment (calebbourg) [reply entity_api/src/coaching_session.rs:15]: That's a good idea. I'll give it a try

Comment (calebbourg) [reply entity_api/src/coaching_session.rs:15]: So for each `entity` the `DeriveEntityModel` adds a `Column` enum that has a variant for each of the table columns.  An example would be `coaching_relationship::Column::CoachingRelationshipId`. This, in my mind would be the best candidate to convert into strings or `&str` in order to compare in these `entity_api` functions. Unfortunately, there isn't a good built in trait for converting enums to `&str` or `&'static str`. There are a few crates that provide implementations like [strum](https://github.com/Peternator7/strum) though I think for now I'd like to hold off on adding the dependency and see if this pattern that we're using here stabilizes in our project.

File: entity_api/src/coaching_session.rs
Comment (calebbourg) [reply entity_api/src/coaching_session.rs:121]: We can't really test that here using this methodology as we're not actually testing what the query returns in terms of records. We're testing the SQL generated by the query.

File: entity_api/src/lib.rs
Comment (calebbourg) [reply entity_api/src/lib.rs:21]: I chose `NaiveDate` because `DateTime` is "timezone aware" as stated in their [docs](https://docs.rs/chrono/latest/chrono/). In our case we chose to set the timezone as a separate column on the table as opposed to encoded in the actual timestamps. We should definitely come back and revisit how this is set up and make sure it's what we're looking for. I'd propose considering that worthy of a discussion and a separate PR if that works for you.

File: web/src/controller/coaching_session_controller.rs
Comment (calebbourg) [reply web/src/controller/coaching_session_controller.rs:37]: I have been copying these over from the other controllers as I believe it is relevant since we currently have no concept of authorization. Made sense to me to propagate the comment just so we would have a sense of where we might want authorization. Open to your thoughts there.

File: entity_api/src/coaching_session.rs
Comment (jhodapp) [reply entity_api/src/coaching_session.rs:15]: Sounds good Caleb, thanks for looking into it it a little bit.

File: entity_api/src/coaching_session.rs
Comment (jhodapp) [reply entity_api/src/coaching_session.rs:121]: More a test for the controller level then? I don't want to get hung up on tests but I did want to ask about anything that came up as a question in my mind while reviewing.

File: entity_api/src/lib.rs
Comment (jhodapp) [reply entity_api/src/lib.rs:21]: Ok sounds good. I created a [discussion to bookmark this conversation](https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/discussions/51#discussion-6901596) and discuss at the right time.

File: web/src/controller/coaching_session_controller.rs
Comment (jhodapp) [reply web/src/controller/coaching_session_controller.rs:22]: Bookmarked as a [future discussion](https://github.com/Jim-Hodapp-Coaching/refactor-platform-rs/discussions/51#discussion-6901596).

File: web/src/controller/coaching_session_controller.rs
Comment (jhodapp) [reply web/src/controller/coaching_session_controller.rs:37]: Ok, so it does still apply. Very good.

**Review by jhodapp — APPROVED** [review]
Looks great Caleb, very nice work as always!

**Pull request merged**
